flink-tests/src/test/java/org/apache/flink/test/operators/ExecutionEnvironmentITCase.java

```diff
@@ -66,6 +74,49 @@ public void mapPartition(Iterable<Integer> values, Collector<Integer> out) throw
 		assertEquals(PARALLELISM, resultCollection.size());
 	}
 
+	@Test
+	public void testAccessingBlockingPersistentResultPartition() throws Exception {
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+		env.setParallelism(1);
+
+		DataSet<Tuple2<Long, Long>> input = env.fromElements(new Tuple2<>(1L, 2L));
+
+		DataSet ds = input.map((MapFunction<Tuple2<Long, Long>, Object>) value -> new Tuple2<>(value.f0 + 1, value.f1));
+
+		// specify IntermediateDataSetID
+		AbstractID intermediateDataSetId = new AbstractID();
+
+		// this output branch will be excluded.
+		ds.output(BlockingShuffleOutputFormat.createOutputFormat(intermediateDataSetId))
+			.setParallelism(1);
+
+		ds.collect();
+
+		BlockingPersistentResultPartitionMeta meta = env.getBlockingPersistentResultPartitionMeta();
+
+		// only one cached IntermediateDataSet
+		Assert.assertEquals(1, meta.getResultPartitionDescriptors().size());
+
+		AbstractID intermediateDataSetID = meta.getResultPartitionDescriptors().keySet().iterator().next();
+
+		// IntermediateDataSetID should be the same
+		Assert.assertEquals(intermediateDataSetID, intermediateDataSetID);
+
+		Map<AbstractID, ResultPartitionDescriptor> descriptors = meta.getResultPartitionDescriptors().get(intermediateDataSetID);
+
+		Assert.assertEquals(1, descriptors.size());
+
+		ResultPartitionDescriptor descriptor = descriptors.values().iterator().next();
+
+		Assert.assertTrue(
```

Comment:
Separate conditions should have separate assertions, so that it is visible from a test failure which of the conditions failed.